packages/core/src/actions/smartAccount/internal/sendUserOperation.ts

```diff
@@ -30,16 +34,27 @@ export const _sendUserOperation: <
     throw new ChainNotFoundError();
   }
 
-  const request = deepHexlify(args.uoStruct);
+  const entryPoint = account.getEntryPoint();
+
+  const request = deepHexlify(
+    args.uoStruct
+  ) as UserOperationRequest<TEntryPointVersion>;
   if (!isValidRequest(request)) {
     throw new InvalidUserOperationError(args.uoStruct);
   }
 
+  request.signature = await account.signUserOperationHash(
+    entryPoint.version === "0.6.0"
+      ? entryPoint.getUserOperationHash(
+          request as UserOperationRequest<"0.6.0">
+        )
+      : entryPoint.getUserOperationHash(
+          request as UserOperationRequest<"0.7.0">
+        )
```

Comment:
just tested this locally and you should be able to just do:
```suggestion
    entryPoint.getUserOperationHash(request)
```

Reply:
bump